main/external-player.js

```diff
@@ -0,0 +1,65 @@
+module.exports = {
+  spawn,
+  kill,
+  checkInstall
+}
+
+var cp = require('child_process')
+var vlcCommand = require('vlc-command')
+
+var log = require('./log')
+var windows = require('./windows')
+
+// holds a ChildProcess while we're playing a video in an external player, null otherwise
+var proc
+
+function checkInstall (path, cb) {
+  // check for VLC if external player has not been specified by the user
+  // otherwise assume the player is installed
+  if (path == null) return vlcCommand((err) => cb(!err))
+  process.nextTick(() => cb(true))
+}
+
+function spawn (path, url) {
+  if (path != null) return _spawn(path, [url])
+
+  // Try to find and use VLC if external player is not specified
+  vlcCommand(function (err, vlcPath) {
+    if (err) return windows.main.dispatch('externalPlayerNotFound')
+    var args = ['--play-and-exit', '--video-on-top', '--no-video-title-show', '--quiet', url]
+    _spawn(vlcPath, args)
+  })
+}
+
+function kill () {
+  if (!proc) return
+  log('Killing external player, pid ' + proc.pid)
+  proc.kill('SIGKILL') // kill -9
+  proc = null
+}
+
+function _spawn (path, args) {
```

Comment:
I'd call this `spawnExternal`. Makes sense that it's not exported. We aren't using underscores for private functions.